reconcile/cli.py

```diff
@@ -417,112 +425,61 @@ def register_faulthandler(fileobj=sys.__stderr__):
         )
 
 
-def run_integration(func_container, ctx, *args, **kwargs):
+def run_integration(
+    func_container: Union[ModuleType, QontractReconcileIntegration],
+    ctx,
+    *args,
+    **kwargs,
+):
     register_faulthandler()
+    dump_schemas_file = ctx.get("dump_schemas_file")
     try:
-        int_name = func_container.QONTRACT_INTEGRATION.replace("_", "-")
-        running_state = RunningState()
-        running_state.integration = int_name
-    except AttributeError:
-        sys.stderr.write("Integration missing QONTRACT_INTEGRATION.\n")
-        sys.exit(ExitCodes.ERROR)
-
-    unleash_feature_state = get_feature_toggle_state(int_name)
-    if not unleash_feature_state:
-        logging.info("Integration toggle is disabled, skipping integration.")
-        sys.exit(ExitCodes.SUCCESS)
-
-    dry_run = ctx.get("dry_run", False)
-    early_exit_compare_sha = ctx.get("early_exit_compare_sha")
+        if isinstance(func_container, QontractReconcileIntegration):
+            integration = func_container
+        elif isinstance(func_container, ModuleType):
+            integration = ModuleBasedQontractReconcileIntegration(func_container)
+        else:
+            raise Exception(f"Unknown integration type {type(func_container)}")
```

Comment:
Please raise a dedicated exception (e.g., `UnknownIntegrationTypeError`) instead of `Exception`

Reply:
done